packages/block-library/src/image/image.js

```diff
@@ -365,6 +370,18 @@ export default function Image( {
 		availableUnits: [ 'px' ],
 	} );
 
+	const lightboxSetting = useSetting( 'lightbox' );
+
+	const lightboxChecked = useMemo( () => {
+		// If the lightbox is set in the block attributes, use that.
+		if ( lightbox?.enabled ) return true;
+
+		// If the lightbox is NOT set in the block attributes AND it IS enabled in
+		// the settings, use that.
+		if ( ! lightbox && lightboxSetting?.enabled ) return true;
+		return false;
+	}, [ lightbox, lightboxSetting ] );
```

Comment:
> `<!-- { "lightbox": true } -->`
> Unexpectedly, this does not enable the lightbox.

This didn't enable the lightbox because I hadn't added support for that syntax yet. I wasn't sure how to do it last week and I determined with some other folks to add it later, but just took a look again and believe I got it working in [this commit](https://github.com/WordPress/gutenberg/pull/54071/commits/7f2f70fc5bd7c2ed3e9f18d47a09e1d118a8a891), with a follow up [here](https://github.com/WordPress/gutenberg/pull/54071/commits/6da3545848524cf9e4c9f990a9c12f14a2278860).

>Moving on to theme.json settings. This example confuses me:
>
>// theme.json
>```
> {
>	"lightbox": {
>		"enabled": false,
>		"allowEditing": true
>	}
>}
> ```
> If I had this in my theme.json settings. What does it mean? Is this a valid combination? At best it's ambiguous; at worst, invalid.

This is indeed a valid combination. The intent here is for `allowEditing` to define whether the UI controls show up, and for `enabled` to determine whether the lightbox behavior should work or not.

> If .enabled is actually the initial state the lightbox value for all images, we should just name it .initial or .default and pass an attribute-shaped thing as the value.

This sounds good to me, but I'll tag in @youknowriad @ramonjd @andrewserong, as they were part of a discussion to define the current structure.

The subtlety here, which I'm not sure is clear, is that we want the lightbox to be _disabled_ by default, but we still want the _UI_ to appear, and also be toggleable. I could see the following structure maybe working:

```
{
	"lightbox": {
		"allowEditing": true
		"default": {
			"enabled": false
		}
	}
}
```

We'll also likely eventually (probably sooner rather than later) need to add support for an `animation` property.

Reply:
Here I'm shooting a follow up on this. @ajlende, would you be alright with the proceeding with the structure as is with `enabled` and `allowEditing`? (Summarized below)

Note: Using `lightbox: true` will both show the lightbox in the UI and enable the lightbox functionality on the frontend by default. Setting `lightbox: false` will remove the lightbox from the UI and disable the lightbox functionality.

**Block Level Settings Syntax**
```
"settings": {
	"blocks": {
		"core/image": {
			"lightbox": {
				"enabled": boolean
				"allowEditing": boolean
			}
		}
	}
}
```
```
"settings": {
	"blocks": {
		"core/image": {
			"lightbox": boolean
		}
	}
}
```

**Top Level Settings Syntax**
```
"settings": {
	"lightbox": {
		"enabled": boolean
		"allowEditing": boolean
	}
}
```
```
"settings": {
	"lightbox": boolean
}
```